tests/integration/test_with_flow.py

```diff
@@ -15,30 +15,52 @@ def encode(self, **kwargs):
 def test_traversal_path():
```

Comment:
These tests should not be `intergation` tests but `unittests`.  Integration tests should be simple

Reply:
I'll move them then

Reply:
I agree, and the tests should be improved in general here. But I think this is out of scope for this PR, here I just wanted to make the code conform to the template + do some necessary fixes